### ISSUE TITLE
[a11y]: Fix issues in IBM Equal Access Karma accessibility tests in React

### ISSUE THREAD

**Issue by jeffchew** [carbon-design-system/carbon-for-ibm-dotcom#7885]
### Description

Currently the a11y tests running are producing a number of accessibility errors. Example output:

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/runs/4511730806?check_suite_focus=true

Engineering team decided to create an upstream feature branch, where the developers will swarm on fixes on this branch. We should not open a PR until all issues are addressed to avoid unnecessary CI checks. We should be checking against the Actions tab for test results while we are working on this.

### Component(s) impacted

Number of components in React

### Carbon for IBM.com version

Canary

### Severity

Severity 3 = The problem is visible or noticeable to users but does not impede the usability or functionality. Affects minor functionality, has a workaround.

### Application/website

Carbon for IBM.com React

### Package

@carbon/ibmdotcom-react

### CodeSandbox example

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/runs/4511730806?check_suite_focus=true

### Steps to reproduce the issue (if applicable)

View the actions tab and look under `Run a11y compliance tests`.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

**Comment by kennylam** [comment]
Duplicate of #8459
